src/libsystemd-network/network-internal.c

```
@@ -131,6 +131,76 @@ int deserialize_in6_addrs(struct in6_addr **ret, const char *string) {
         return size;
 }
 
+int serialize_dnr(FILE *f, const ResolverData *resolvers, bool *with_leading_space) {
+        int r;
+
+        bool _space = false;
+        if (!with_leading_space)
+                with_leading_space = &_space;
+
+        int n = 0;
+        LIST_FOREACH(resolvers, i, resolvers) {
+                _cleanup_strv_free_ char **names = NULL;
+                r = dns_resolvers_to_dot_strv(resolvers, &names);
+                if (r < 0)
+                        return -ENOMEM;
```

why not propagate the original error `r` here?